typed-racket-lib/typed-racket/rep/object-rep.rkt

```diff
@@ -6,37 +6,37 @@
 ;; See "Logical Types for Untyped Languages" pg.3
 
 (require "../utils/utils.rkt"
+         racket/match
          "rep-utils.rkt"
          "core-rep.rkt"
          "free-variance.rkt"
          (env mvar-env)
          (contract-req))
 
-(provide -id-path)
+(provide -id-path name-ref=?)
+
+(def-pathelem CarPE () [#:singleton -car])
+(def-pathelem CdrPE () [#:singleton -cdr])
+(def-pathelem SyntaxPE () [#:singleton -syntax-e])
+(def-pathelem ForcePE () [#:singleton -force])
+(def-pathelem FieldPE () [#:singleton -field])
 
-(def-pathelem CarPE () #:base)
-(def-pathelem CdrPE () #:base)
-(def-pathelem SyntaxPE () #:base)
-(def-pathelem ForcePE () #:base)
 ;; t is always a Name (can't put that into the contract b/c of circularity)
 (def-pathelem StructPE ([t Type?] [idx natural-number/c])
```

Comment:
The comment is actually just wrong if I remember correctly (during a refactoring I pushed Name up a level and Names weren't the only thing to enter that field).